caffe2/python/operator_test/weighted_sum_test.py

```diff
@@ -4,20 +4,21 @@
 from __future__ import unicode_literals
 
 from caffe2.python import core
-from hypothesis import given
 import caffe2.python.hypothesis_test_util as hu
 import hypothesis.strategies as st
 import numpy as np
+import caffe2.python.serialized_test.serialized_test_util as serial
 
 
-class TestWeightedSumOp(hu.HypothesisTestCase):
+class TestWeightedSumOp(serial.SerializedTestCase):
 
-    @given(n=st.integers(5, 8), m=st.integers(1, 1),
-           d=st.integers(2, 4), grad_on_w=st.booleans(),
-           **hu.gcs_cpu_only)
+    @serial.given_and_seeded(
```

Comment:
Can we don't change given, but have another separate wrapper, called `serialize_with_seed()`

Reply:
We had previously tried to do this, but it proved to be tricky because of the following
1. hypothesis.given annotates a test with a bunch of variables
https://hypothesis.readthedocs.io/en/latest/_modules/hypothesis/core.html
2. when we define a decorator outside of hypothesis, we don't have access to those variables for modification, because we don't assume we know anything about the wrapped function